internal/peer/node/start.go

```diff
@@ -185,6 +186,23 @@ func (c custodianLauncherAdapter) Stop(ccid string) error {
 }
 
 func serve(args []string) error {
+	logger.Infof("Starting %s", version.GetInfo())
+
+	// Info logging for peer config, includes core.yaml settings and environment variable overrides
+	allSettings := viper.AllSettings()
+	settingsYaml, err := yaml.Marshal(allSettings)
+	if err != nil {
+		return err
+	}
+	logger.Infof("Peer config with combined core.yaml settings and environment variable overrides:\n%s", settingsYaml)
+
+	// Debug logging for peer environment variables
+	logger.Debugf("Environment variables:")
```

Comment:
Sorry Dave, a nit: any specific reason why we use debug for this but info for the previous?

Reply:
I was debating this myself. In my opinion the exact peer config should be easily knowable by every user and support engineer hence Info level. This is consistent with what orderer node provides but for some reason peer never provided it.
The wider set of environment variables may be less important and potentially could leak sensitive information about the operating environment hence Debug level.
However, showing the environment variables will help a user or support engineer know whether a certain config setting came from core.yaml or from an environment variable override, so it is valuable information when troubleshooting a peer config.
I could go either way, WDYT?